monai/transforms/io/dictionary.py

```diff
@@ -127,8 +133,18 @@ def __call__(self, data, reader: Optional[ImageReader] = None):
 
         """
         d = dict(data)
+        image_shapes = []
+        # From python 3.6 and up, dictionaries are ordered, so we can assume that the dict orders we get
+        # are all in the same order, unless the user goes out of their way to shuffle them.
+        first_path = list(d.values())[0]
+        # Keep a copy of the original input dict around in case we need to log it
+        input_dict = dict(data)
+
         for key, meta_key, meta_key_postfix in self.key_iterator(d, self.meta_keys, self.meta_key_postfix):
             data = self._loader(d[key], reader)
+            # _loader returns a tuple of (ndarray, dict), we want the ndarray.shape
+            image_shapes.append(data[0].shape)
```

Comment:
if `image_only` is `True`, then data will be just the ndarray, not a tuple. So you probably want something like:
```suggestion
            image_shapes.append(data[0].shape if isinstance(data, tuple) else data.shape)
```